software/controller/lib/core/breath_detection.h

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2020-2022, RespiraWorks
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#pragma once
+
+#include <optional>
+
+#include "units.h"
+#include "vars.h"
+
+// Sensor readings etc that are used by breath detection.
+struct BreathDetectionInputs {
+  // Current patient volume, corrected for sensor zero-point drift.
+  Volume patient_volume;
+  // inflow - outflow, corrected for sensor zero-point drift.
+  VolumetricFlow net_flow;
+};
+
+// Provides a class that is used to detecting inspiratory effort.
+class BreathDetection {
```

Comment:
Is this class now unit testable?
If so, could at least make a stump of a test and leave the implementation for later in TODO. To at least capture it in coverage statistics.